rkt/metadata_service.go

```diff
@@ -75,6 +71,109 @@ func init() {
 	cmdMetadataService.Flags.BoolVar(&flagNoIdle, "no-idle", false, "exit when last container is unregistered")
 }
 
+type mdsPod struct {
+	uuid     types.UUID
+	ip       string
+	manifest *schema.PodManifest
+	apps     map[string]*schema.ImageManifest
+}
+
+type podStore struct {
+	byIP   map[string]*mdsPod
+	byUUID map[types.UUID]*mdsPod
+	mutex  sync.Mutex
```

Comment:
But I don't want Lock/Unlock methods to become part of the podStore interface. The whole point is that the interface is a set of specific accessors.